File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        if (root == nullptr) return vector<vector<int>> ();
        vector<vector<int>> ans;
        queue<TreeNode*> queue;
        queue.push(root);
        while(!queue.empty()){
            vector<int> temp;
            int n = queue.size();
            for (int i =0 ; i < n; i++){
                TreeNode* node = queue.front();
                queue.pop();
                temp.push_back(node->val);
                if (node->left != nullptr)  queue.push(node->left);
                if (node->right != nullptr)  queue.push(node->right);

            }
            ans.push_back(temp);
        }

    return ans;}
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ans = new ArrayList<>(1);
        dfs(root, ans, 0);
        return ans;
    }

    void dfs(TreeNode root, List<List<Integer>> ans, int depth) {
        if (root == null)
            return;
        if (ans.size() == depth)
            ans.add(new ArrayList<>());
        ans.get(depth).add(root.val);
        dfs(root.left, ans, depth + 1);
        dfs(root.right, ans, depth + 1);

    }
}